### ISSUE TITLE
Setting poses via script

### ISSUE THREAD

**Issue by timetosmile** [ValveSoftware/steamvr_unity_plugin#319]
Hi there,

I updated now to steamVR v. 2.2 and using Unity v. 2018.3.4f1.
It is really nice and comfortable to create a pose and attache it to an interactable object. 

In my case I need to create an area around my menu where my hand need to be in a pointing pose. 
So I created a Transform with a collider and the skeleton_Poser component. There I defined my pose. 

![pointing pose_2](https://user-images.githubusercontent.com/12845370/52622477-392bca80-2eaa-11e9-8b13-4aa872ae1fdd.png)

In my UINearField component I am looking for a OnTriggerEnter() and want then to change my pose by 
`leftHand.skeleton.BlendToPoser(skeletonPoser);` (same for the right hand)

For my right hand this work very well but my left hand always looks like it got smashed by a big hammer.  

![pointing pose](https://user-images.githubusercontent.com/12845370/52622441-287b5480-2eaa-11e9-8f07-9c67b0683fd2.png)

I tried it with different poses but always the same result. 
I looked into skeleten_Poser and saw that you safe the blendSnapshots for both hands so I don't really unterstand why my left hand is always that broken. 

My 2. question to this topic is, how can I get back to my grabing pose? 
I defined a second pose in the skeleton_Poser component but with `leftHand.skeleton.BlendToPoser(skeletonPoser);` 
I can only set the main pose. I could use `leftHand.skeleton.BlendToPoser(skeletonPoser.GetPoseByIndex(X));` but this only returns a pose where I need a poser. 

What do I miss or is it not intended to set a pose outside an interactable object?

**Comment by zite** [comment]
you can blend back to the skeleton using `leftHand.skeleton.BlendToSkeleton();`

I'll have to look into the one hand being broken more and get back to you.

**Comment by zulubo** [comment]
That's an odd bug. Can you try using that pose on an interactable and see if it breaks as well? If so, there's something wrong with the poses you're making. If not, there's something wrong with the way you're applying the pose. Also, have you tried some of the built in poses, or just ones you've authored?

Also, when you want to blend to multiple poses on a single SkeletonPoser, you just need to add a blending behaviour and set the weight of it via code. Seems like that's not what you're trying to do, but it's good to know.

**Comment by timetosmile** [comment]
Hey there, 
> you can blend back to the skeleton using `leftHand.skeleton.BlendToSkeleton();` 
that works perfectly. Thank you. 

So, my code is now like this 
`
public SteamVR_Skeleton_Poser skeletonPoser;

        private void OnTriggerEnter(Collider other)
        {
            if (other.transform.IsChildOf(leftHand.transform))
                 leftHand.skeleton.BlendToPoser(skeletonPoser);
            
            if (other.transform.IsChildOf(rightHand.transform))
                 rightHand.skeleton.BlendToPoser(skeletonPoser);       
        }

        private void OnTriggerExit(Collider other)
        {
            if (other.transform.IsChildOf(leftHand.transform))
                 leftHand.skeleton.BlendToSkeleton();
            
            if (other.transform.IsChildOf(rightHand.transform))
                 rightHand.skeleton.BlendToSkeleton();
        }

`

and I tried out different poses and got still a similiar broken hand. After some test a found out, that this happen, when one of the left finger movement enums is something else then *static* .
On the right hand it doesn't matters.
When I got this error and pull then the trigger, then the hand move to his defined pose and then move back to a broken state. 
The broken state of the left hand looks like the left hand tries the pose of the right hand. That's why it looks like "inside-out".

I thought that the movement types free extendet means, that the finger can move between the defined pose and the default pose from before. But this only works for the right hand. 

I am not sure if this is only up to me. Maybe someone can test it now a little bit better. For the moment I can use static fingers and it is enough that way. 

Thank you for your response.

**Comment by zite** [comment]
Thanks for continuing to troubleshoot this with us. Is this true for everything besides static? Or just Free? Free will match the orientation of the finger without the pose so it might make sense if it was going into the pose of the rest of your hand.

**Comment by timetosmile** [comment]
> Thanks for continuing to troubleshoot this with us. Is this true for everything besides static? Or just Free? Free will match the orientation of the finger without the pose so it might make sense if it was going into the pose of the rest of your hand.

Thank you for your work. 

This is true for everything besides static. So this is what I got.

**Comment by gkapellmann** [comment]
So, regarding this question, I got it working as described here, but when I use "BlendToSkeleton()" it seems that my pose is overwritten somewhere, as as soon as I use the "Interactable" script, and change the index of "Hand Animation on Pickup" from 0 to 1, this pose has been changed to the one I setup when using "BlendToPoser(skeletonPoser)". However, if I dont set the poser, the original pose from "Hand Animation on Pickup" appears, it only changes when "BlendToPoser(skeletonPoser)" has been executed at least once.

Is this meant to happen like this?

**Comment by zite** [comment]
Heya this may be fixed in the beta version of the plugin here: https://github.com/ValveSoftware/steamvr_unity_plugin/releases/tag/2.3b1

**Comment by zite** [comment]
Sounds like this was fixed in 2.3.2. Closing.

**Comment by PanosCortese** [comment]
Hi, I had the same issue. Also, my issue was either with the Left or the Right hand (depending on which controller was activating first). I noticed that when I press the system button of Vive controllers, the pose was getting corrected to the expected pose (from this twisted inside out thing). I thought that there was something with the Mirroring. However, I managed to solve it by using a very simplistic solution. I created two identical triggers, one for the Right controller and one for Left Controller. In each trigger, I had a simple script (OnTrigger Enter/Exit), where I change the pose of the respective Controller/Hand.  So, two identical Triggers(Colliders), with the same Skeleton Poser (identical, like when you're copying and pasting the component), but with a different script in each one, where the event of each script changes the pose of a single controller. Hope that my simple solution might help someone stuck, to get around of this annoying issue.
